CRM/Contribute/BAO/ContributionPage.php

```diff
@@ -483,7 +483,7 @@ protected static function getProfileNameAndFields($gid, $cid, $params, $fieldTyp
         $fields = CRM_Core_BAO_UFGroup::getFields($gid, FALSE, CRM_Core_Action::VIEW, NULL, NULL, FALSE, NULL, FALSE, NULL, CRM_Core_Permission::CREATE, NULL);
         foreach ($fields as $k => $v) {
           if (!$groupTitle) {
-            $groupTitle = $v['groupTitle'];
+            $groupTitle = !empty($v['groupDisplayTitle']) ? $v['groupDisplayTitle'] : $v['groupTitle'];
```

Comment:
@samuelsov since we stopped supporting earlier versions of php (maybe 5.x?) we have been able to start using this shorter syntax which is pretty nice
```
 $groupTitle = $v['groupDisplayTitle'] ?? $v['groupTitle'];
```

Reply:
(it's called a Null coalescing operator - cachy huh?)

Reply:
Here wouldn't you want it to be more like `(isset($v['groupDisplayTitle']) && $v['groupDisplayTitle'] !== '') ? $v['groupDisplayTitle'] : $v['groupTitle'])` because '0' is a legitimate, albeit weird, title. And if using `??` it also would then end up using the empty string even if they left the field blank.

But I'm wondering if you can just do `$v['groupDisplayTitle']` because if you look at how that's defined in formatUFField it should never be blank and already falls back to groupTitle. (And it has the same `0` problem but whatever.)

Reply:
@demeritcowboy you were right, getFields go through formatUFField which takes care of setting the groupDisplayTitle properly... PR updated